dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java

```diff
@@ -81,70 +82,73 @@ final public PublishStatus publish ( PublisherConfig config, PublishStatus statu
                 // will return true after that always.
                 final boolean bundleExists = BundlerUtil.bundleExists(config);
 
-                // Run bundlers
-
-                if (config.isStatic()) {
-                    //If static we just want to save the things that we need,
-                    // at this point only the id, static and operation.
-                	PublisherConfig pcClone = new PublisherConfig();
-                	pcClone.setId(config.getId());
-                	pcClone.setStatic(true);
-                	pcClone.setOperation(config.getOperation());
-                    Logger.info(this, "Writing bundle.xml file");
-                	BundlerUtil.writeBundleXML( pcClone, output );
-                } else {
-                    Logger.info(this, "Writing bundle.xml file");
-                    BundlerUtil.writeBundleXML( config, output );
-                }
+                try (BundleOutput output = publisher.createBundleOutput()){
+                    status.addOutput(output);
+                    // Run bundlers
+
+                    if (config.isStatic()) {
+                        //If static we just want to save the things that we need,
+                        // at this point only the id, static and operation.
+                        PublisherConfig pcClone = new PublisherConfig();
+                        pcClone.setId(config.getId());
+                        pcClone.setStatic(true);
+                        pcClone.setOperation(config.getOperation());
+                        Logger.info(this, "Writing bundle.xml file");
```

Comment:
This two lines could be out of the `if` block and the `else` block can be removed.

Reply:
really this line are not exactly the same because it pass different parameters to 

BundlerUtil.writeBundleXML(pcClone, output);

Really I don't understand why it crate the pcClone